include/nuttx/audio/audio.h

```diff
@@ -534,6 +536,10 @@ typedef CODE void (*audio_callback_t)(FAR void *priv, uint16_t reason,
 struct audio_lowerhalf_s;
 struct audio_ops_s
 {
+  /* This method is called when the related device file is opened. */
+
+  CODE int (*setup)(FAR struct audio_lowerhalf_s *dev, int opencnt);
```

Comment:
should we move the new callback to last to keep the source code compatibility, and revert the change from the individual driver.